Implements back end game logic

PThompson284 opened the pull request:
Implements the rules of the game and allows you to play
#19 #20

DEVELOPMENT BRANCH PR

PThompson284 committed:
Basic structure of game and movement logic

PThompson284 committed:
Added ability to move pieces and start of movement validity logic.

PThompson284 committed:
Stylization changes

PThompson284 committed:
Stylization changes

PThompson284 committed:
Fix issue with turns not changing

PThompson284 committed:
Merge branch 'pt_dev' of https://github.com/DavidWellsTheDeveloper/cs414-f18-001-Team.name into pt_dev

PThompson284 committed:
Updated game to receive commands one tile at a time instead of having both tiles of the move known at once.

PThompson284 committed:
Update game logic for movement to handle water, friendly, and enemy units.

PThompson284 committed:
updated game logic and added comments

brian-crane committed:
Adding new JUnit Test for bcrane and Java-fying Game variables

PThompson284 committed:
Added game logic for lion/tiger leaping
Added game logic for win conditions

PThompson284 committed:
Merge branch 'pt_dev' of https://github.com/DavidWellsTheDeveloper/cs414-f18-001-Team.name into pt_dev

PThompson284 committed:
Expand on testing for game logic

PThompson284 committed:
Fix handling of traps (friendly vs enemy)
Expanded on tests

PThompson284 committed:
Expand on test output

PThompson284 committed:
Change how commands are received to be (row, col) instead of (col, row) to be consistent with internal board structure

tmsdbqk commented on server/src/Game.java:
```diff
@@ -0,0 +1,114 @@
+        playerTwo = playerTwoName;
+        turn = "blue";
+
+        for(int i=0; i<9; i++){
```
Suggest changing magic numbers and indices to something more descriptive.

tmsdbqk commented on server/src/Game.java:
```diff
@@ -0,0 +1,114 @@
+                move.selectedCol = col;
+                move.selectedRow = row;
+            }
+            else if(move.validTiles[row][col].equals("*")) { // if row, col is a valid tile to move to
```
What if for some reason the row or column has an index out of bounds of the board?

tmsdbqk commented on server/src/Game.java:
```diff
@@ -0,0 +1,114 @@
+        return (turn.equals("blue") && player.equals(playerOne)) || (turn.equals("red") && player.equals(playerTwo));
+    }
+
+    private void checkIfWinner(){
```
We might want some of these methods to return an integer or flag of some sort, so the server knows when to make an update to the database without having to call the object again. Return -1 if no winner, or the player ID if winner, etc. We can talk about it more.

tmsdbqk commented on server/src/Move.java:
```diff
@@ -0,0 +1,128 @@
+            }
+        }
+        if(selectedRow != -1 && selectedCol != -1) {
+            if(selectedCol <= 5) handleUpdate(selectedRow, selectedCol+1);
```
What do these numbers mean?

brian-crane committed:
Changing GameTest to use normal JUnit Annotations and removing GameTest2

brian-crane committed:
Changing GameTest to use normal JUnit Annotations and removing GameTest2

brian-crane committed:
Adding GameUtils class to help visualize board
Adding new JUnit test to make sure you cant move pieces onto existing pieces
Adding print statement to MakeMove method in Game class

brian-crane committed:
Resolving merge conflict with pt_dev by updating sendInput method sig

brian-crane committed:
Merge remote-tracking branch 'remotes/origin/pt_dev' into bc_pt_dev

# Conflicts:
#	server/src/Game.java

brian-crane committed:
Updating Game.java to show move sequence, and to use GameUtils

Also added changes to JUnit test testBlueLionNotAbleToCrossItsOwnPieces to use new sendMove method signature

brian-crane committed:
Merge pull request #40 from DavidWellsTheDeveloper/bc_pt_dev

Bc pt dev
